contributors/design-proposals/pod-injection-policy.md

```diff
@@ -0,0 +1,662 @@
+  * [Use Cases](#use-cases)
+    * [Loose coupling between services and their consumers](#loose-coupling-between-services-and-their-consumers)
+  * [Proposed Changes](#proposed-changes)
+    * [PodInjectionPolicy API object](#podinjectionpolicy-api-object)
+      * [Validations](#validations)
+    * [AdmissionControl Plug-in: PodInjectionPolicy](#admissioncontrol-plug-in-podinjectionpolicy)
+      * [Behavior](#behavior)
+      * [Optional Behavior with implicit injections](#optional-behavior-with-implicit-injections)
+  * [Examples](#examples)
+    * [Simple Pod Spec Example](#simple-pod-spec-example)
+    * [Pod Spec with `ConfigMap` Example](#pod-spec-with-`configmap`-example)
+    * [ReplicaSet with Pod Spec Example](#replicaset-with-pod-spec-example)
+    * [Multiple PodInjectionPolicy Example](#multiple-podinjectionpolicy-example)
+    * [Conflict Example](#conflict-example)
+
+
+## Abstract
+
+Describes a policy resource that allows for the loose coupling of a Pod's
+definition from additional runtime requirements for that Pod. For example,
+mounting of Secrets, or setting additional environment variables,
+may not be known at Pod deployment time, but may be required at Pod creation
+time.
+
+## Motivation
+
+Consuming a service involves more than just connectivity. In addition to
+coordinates to reach the service, credentials and non-secret configuration
+parameters are typically needed to use the service.  The primitives for this
+already exist, but a gap exists where loose coupling is desired: it should be
+possible to inject pods with the information they need to use a service on a
+service-by-service basis, without the pod authors having to incorporate the
+information into every pod spec where it is needed.
+
+## Constraints and Assumptions
+
+1.  Future work might require new mechanisms to be made to work with existing
+    controllers such as deployments and replicasets that create pods. Existing
+    controllers that create pods should recreate their pods when a new Pod Injection
+    Policy is added that would effect them.
+
+## Use Cases
+
+1.  As a user, I want to be able to describe a way that pods should be injected
+    with the information to consume a particular service. This would allow me
+    to concisely model the information about how the service
+    should be consumed without altering every consuming pod spec.
+
+### Loose coupling between services and their consumers
+
+Using a Pod Injection Policy allows pod template authors to not have to explicitly
+set information for every pod. This way authors of pod templates consuming a
+specific service do not need to know all the details about that service.
+
+The pods chosen for injection will be based off the policy's `Selector`.
+
+## Proposed Changes
+
+### PodInjectionPolicy API object
+
+This resource is alpha. The policy itself is immutable. The API group will be
+called `podinjection` and the version is `v1alpha1`.
```

Comment:
group = package, resource = class name.  There are lots of important classes that are the only type in their package.  

I actually kind of expected this to be in the "apps" group.  Since the point of this is to create abstractions for apps (which are more than just single pods) that are in common.